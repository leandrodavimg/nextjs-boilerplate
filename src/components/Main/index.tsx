/* eslint-disable react/react-in-jsx-scope */
// eslint-disable-next-line react/react-in-jsx-scope
const Main = ({ title = 'React AvanÃ§ado' }) => (
  <main>
    <h1>{title}</h1>
  </main>
)

export default Main
