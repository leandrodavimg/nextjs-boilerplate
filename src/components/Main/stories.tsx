import { Story, Meta } from '@storybook/react/types-6-0'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Passando pelo default!!!'
  }
} as Meta

export const Basic: Story = (args) => <Main {...args} />
// Basic.args = {
//   title: 'React Leandro Teste!!!'
// }
