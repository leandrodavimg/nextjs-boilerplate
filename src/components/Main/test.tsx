import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the reading', () => {
    // eslint-disable-next-line react/react-in-jsx-scope
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /React avan√ßado!!!/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
