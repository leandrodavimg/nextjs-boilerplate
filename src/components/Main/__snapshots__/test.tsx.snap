// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the reading 1`] = `
<main>
  <h1>
    React avan√ßado!!!
  </h1>
</main>
`;
